### PR TITLE
Document older version numbers and why they might be used

### LINKED ISSUE #2359: How to select a BOM version?
### Describe your use-case which is not covered by existing documentation.

https://github.com/jenkinsci/bom#usage suggests:

```
After selecting your plugin’s LTS baseline:

<jenkins.version>2.387.3</jenkins.version>

just import the [latest BOM](https://github.com/jenkinsci/bom/releases) from that line:

<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>io.jenkins.tools.bom</groupId>
            <artifactId>bom-2.387.x</artifactId>
            <version>…</version>
            <scope>import</scope>
            <type>pom</type>
        </dependency>
    </dependencies>
</dependencyManagement>
```

`artifactid` is clear - select the version corresponding to the `jenkins.version` value. What about `version`? Is the suggestion to go to https://github.com/jenkinsci/bom/releases and select the latest release without regard for the jenkins.version?

I found this confusing. I think  a tweak would help greatly!

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

_No response_

### PR DESCRIPTION
**Pull request by MarkEWaite**
## Document older version numbers and why they might be used

Fix #2359 by documenting the plugin bill of materials versions that are no longer being updated and by describing some of the cases where those plugin BOM versions might be the right choice.  

### Testing done

Reviewed the text and the example plugin bill of materials version numbers.

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,95 @@ You would not use it:
...
+        </dependency>
+    </dependencies>
+</dependencyManagement>
+```
+
+### First LTS to support Java 11 - 2.361.x
+
+Choose Jenkins 2.361.4 if the plugin supports Java 11 but needs to remain on an older release.
+Jenkins 2.361.4 was released in November of 2022.
+After November 2023, newer Jenkins versions should be chosen rather than Jenkins 2.361.x.
```
Comment (lemeurherve) [inline README.md:121]: Maybe tell here what's the first next version to use?

**Review by lemeurherve — COMMENTED** [review]
LGTM, small optional suggestion.

File: README.md (hunk 1 above)
Comment (MarkEWaite) [reply README.md:121]: Thanks for the review.  I was intentionally not listing the next version to choose in hopes that this text would live longer before it needed to be revised.  The intent of the phrasing was to lead them to follow the instructions from "choosing a version" and select a base release that is roughly one year old.

**Review by timja — COMMENTED** [review]
this seems like too much maintenance, is there that much demand for old lines?

**Comment by MarkEWaite** [comment]
> this seems like too much maintenance, is there that much demand for old lines?

A good question.  The maintenance is easier for me personally than finding the terminal version of a line by other means.  It could also be as simple as a table of older versions and the last release of that older version.  I've been using the table in the issue report  #2359 while adding Java 21 test configurations in the top 250 plugins.

**Comment by timja** [comment]
> > this seems like too much maintenance, is there that much demand for old lines?
> 
> A good question. The maintenance is easier for me personally than finding the terminal version of a line by other means. It could also be as simple as a table of older versions and the last release of that older version. I've been using the table in the issue report #2359 while adding Java 21 test configurations in the top 250 plugins.

I think a table would be simpler, there's quite a lot of content for generally niche content I think.

**Comment by MarkEWaite** [comment]
> I think a table would be simpler, there's quite a lot of content for generally niche content I think.

Thanks.  I've replaced most of the text with a comment that links to the developer documentation, a table of the most commonly used older versions, and a link to the complete list of all versions.

**Review by timja — APPROVED** [review]

**Pull request merged**